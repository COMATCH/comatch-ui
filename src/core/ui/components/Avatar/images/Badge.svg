<?xml version="1.0" encoding="UTF-8"?>
<svg width="20px" height="24px" viewBox="0 0 20 24" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: sketchtool 53.2 (72643) - https://sketchapp.com -->
    <title>E7013BA0-2C2D-4043-A2FC-9860EC8294A5</title>
    <desc>Created with sketchtool.</desc>
    <g id="UI" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="project_details" transform="translate(-230.000000, -604.000000)" fill-rule="nonzero">
            <g id="card4" transform="translate(133.000000, 579.000000)">
                <g id="silver_status" transform="translate(97.000000, 25.000000)">
                    <path d="M13.135828,11.252948 L19.1724731,0.709595376 C19.1722907,0.521214988 19.097872,0.340623225 18.9655882,0.207548259 C18.8333045,0.0744732922 18.6539917,-0.000184055049 18.4670968,-2.85495643e-13 L0.746666667,-2.85495643e-13 C0.559533238,-0.00018429008 0.380011753,0.0746634145 0.247688373,0.208038498 C0.115364993,0.341413581 0.0411074857,0.522361898 0.0412903226,0.710982659 L6.70623656,10.8166474 L6.74890323,10.8811561 L13.135828,11.252948 Z" id="Shape" fill="#82B3BB"></path>
                    <path d="M6.67870968,10.8617341 L6.70623656,10.8166474 L13.488172,-3.39514372e-07 L0.746666667,-2.85495643e-13 C0.559533238,-0.00018429008 0.380011753,0.0746634145 0.247688373,0.208038498 C0.115364993,0.341413581 0.0411074857,0.522361898 0.0412903226,0.710982659 L6.70623656,10.8166474 L6.74890323,10.8811561" id="Shape" fill="#05677A"></path>
                    <ellipse id="Oval" fill="#C0C0C0" cx="9.61169892" cy="17.1024277" rx="6.78606452" ry="6.84"></ellipse>
                    <path d="M13.3539785,16.1986127 C13.3445193,16.282996 13.3024949,16.3602378 13.2369892,16.4136416 L11.603957,17.9972254 L11.9907097,20.2342197 C11.9945625,20.2638801 11.9961728,20.2937929 11.9955269,20.3236994 C11.9974259,20.3804655 11.9807398,20.4362842 11.948043,20.4825434 C11.9162927,20.5266282 11.864344,20.5512382 11.8104086,20.5477457 C11.7468681,20.5458888 11.6849278,20.5272308 11.6307957,20.4936416 L9.61101075,19.4379191 L7.59122581,20.4936416 C7.53670869,20.5263557 7.47499895,20.5449443 7.4116129,20.5477457 C7.3564475,20.5516417 7.30308127,20.5270966 7.26984946,20.4825434 C7.23837003,20.4358559 7.22290283,20.3800725 7.22580645,20.3236994 C7.22686506,20.2937092 7.22985342,20.2638198 7.23475269,20.2342197 L7.62150538,17.9972254 L5.9815914,16.4136416 C5.91903881,16.3584666 5.87906053,16.28183 5.86941935,16.1986127 C5.86941935,16.0880925 5.95337634,16.0187283 6.12129032,15.9905202 L8.3791828,15.663815 L9.39148387,13.6279769 C9.44791398,13.505896 9.52154839,13.4441618 9.61169892,13.4441618 C9.70184946,13.4441618 9.7747957,13.505896 9.83191398,13.6279769 L10.8442151,15.663815 L13.1021075,15.9905202 C13.2700215,16.0187283 13.3539785,16.0880925 13.3539785,16.1986127 L13.3539785,16.1986127 Z" id="Shape" fill="#FFFFFF"></path>
                </g>
            </g>
        </g>
    </g>
</svg>
