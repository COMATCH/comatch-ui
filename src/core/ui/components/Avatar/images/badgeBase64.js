/* eslint-disable max-len */
export const badgeBase64 =
    'data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz4KPHN2ZyB3aWR0aD0iMjBweCIgaGVpZ2h0PSIyNHB4IiB2aWV3Qm94PSIwIDAgMjAgMjQiIHZlcnNpb249IjEuMSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayI+CiAgICA8IS0tIEdlbmVyYXRvcjogc2tldGNodG9vbCA1My4yICg3MjY0MykgLSBodHRwczovL3NrZXRjaGFwcC5jb20gLS0+CiAgICA8dGl0bGU+RTcwMTNCQTAtMkMyRC00MDQzLUEyRkMtOTg2MEVDODI5NEE1PC90aXRsZT4KICAgIDxkZXNjPkNyZWF0ZWQgd2l0aCBza2V0Y2h0b29sLjwvZGVzYz4KICAgIDxnIGlkPSJVSSIgc3Ryb2tlPSJub25lIiBzdHJva2Utd2lkdGg9IjEiIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCI+CiAgICAgICAgPGcgaWQ9InByb2plY3RfZGV0YWlscyIgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoLTIzMC4wMDAwMDAsIC02MDQuMDAwMDAwKSIgZmlsbC1ydWxlPSJub256ZXJvIj4KICAgICAgICAgICAgPGcgaWQ9ImNhcmQ0IiB0cmFuc2Zvcm09InRyYW5zbGF0ZSgxMzMuMDAwMDAwLCA1NzkuMDAwMDAwKSI+CiAgICAgICAgICAgICAgICA8ZyBpZD0ic2lsdmVyX3N0YXR1cyIgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoOTcuMDAwMDAwLCAyNS4wMDAwMDApIj4KICAgICAgICAgICAgICAgICAgICA8cGF0aCBkPSJNMTMuMTM1ODI4LDExLjI1Mjk0OCBMMTkuMTcyNDczMSwwLjcwOTU5NTM3NiBDMTkuMTcyMjkwNywwLjUyMTIxNDk4OCAxOS4wOTc4NzIsMC4zNDA2MjMyMjUgMTguOTY1NTg4MiwwLjIwNzU0ODI1OSBDMTguODMzMzA0NSwwLjA3NDQ3MzI5MjIgMTguNjUzOTkxNywtMC4wMDAxODQwNTUwNDkgMTguNDY3MDk2OCwtMi44NTQ5NTY0M2UtMTMgTDAuNzQ2NjY2NjY3LC0yLjg1NDk1NjQzZS0xMyBDMC41NTk1MzMyMzgsLTAuMDAwMTg0MjkwMDggMC4zODAwMTE3NTMsMC4wNzQ2NjM0MTQ1IDAuMjQ3Njg4MzczLDAuMjA4MDM4NDk4IEMwLjExNTM2NDk5MywwLjM0MTQxMzU4MSAwLjA0MTEwNzQ4NTcsMC41MjIzNjE4OTggMC4wNDEyOTAzMjI2LDAuNzEwOTgyNjU5IEw2LjcwNjIzNjU2LDEwLjgxNjY0NzQgTDYuNzQ4OTAzMjMsMTAuODgxMTU2MSBMMTMuMTM1ODI4LDExLjI1Mjk0OCBaIiBpZD0iU2hhcGUiIGZpbGw9IiM4MkIzQkIiPjwvcGF0aD4KICAgICAgICAgICAgICAgICAgICA8cGF0aCBkPSJNNi42Nzg3MDk2OCwxMC44NjE3MzQxIEw2LjcwNjIzNjU2LDEwLjgxNjY0NzQgTDEzLjQ4ODE3MiwtMy4zOTUxNDM3MmUtMDcgTDAuNzQ2NjY2NjY3LC0yLjg1NDk1NjQzZS0xMyBDMC41NTk1MzMyMzgsLTAuMDAwMTg0MjkwMDggMC4zODAwMTE3NTMsMC4wNzQ2NjM0MTQ1IDAuMjQ3Njg4MzczLDAuMjA4MDM4NDk4IEMwLjExNTM2NDk5MywwLjM0MTQxMzU4MSAwLjA0MTEwNzQ4NTcsMC41MjIzNjE4OTggMC4wNDEyOTAzMjI2LDAuNzEwOTgyNjU5IEw2LjcwNjIzNjU2LDEwLjgxNjY0NzQgTDYuNzQ4OTAzMjMsMTAuODgxMTU2MSIgaWQ9IlNoYXBlIiBmaWxsPSIjMDU2NzdBIj48L3BhdGg+CiAgICAgICAgICAgICAgICAgICAgPGVsbGlwc2UgaWQ9Ik92YWwiIGZpbGw9IiNDMEMwQzAiIGN4PSI5LjYxMTY5ODkyIiBjeT0iMTcuMTAyNDI3NyIgcng9IjYuNzg2MDY0NTIiIHJ5PSI2Ljg0Ij48L2VsbGlwc2U+CiAgICAgICAgICAgICAgICAgICAgPHBhdGggZD0iTTEzLjM1Mzk3ODUsMTYuMTk4NjEyNyBDMTMuMzQ0NTE5MywxNi4yODI5OTYgMTMuMzAyNDk0OSwxNi4zNjAyMzc4IDEzLjIzNjk4OTIsMTYuNDEzNjQxNiBMMTEuNjAzOTU3LDE3Ljk5NzIyNTQgTDExLjk5MDcwOTcsMjAuMjM0MjE5NyBDMTEuOTk0NTYyNSwyMC4yNjM4ODAxIDExLjk5NjE3MjgsMjAuMjkzNzkyOSAxMS45OTU1MjY5LDIwLjMyMzY5OTQgQzExLjk5NzQyNTksMjAuMzgwNDY1NSAxMS45ODA3Mzk4LDIwLjQzNjI4NDIgMTEuOTQ4MDQzLDIwLjQ4MjU0MzQgQzExLjkxNjI5MjcsMjAuNTI2NjI4MiAxMS44NjQzNDQsMjAuNTUxMjM4MiAxMS44MTA0MDg2LDIwLjU0Nzc0NTcgQzExLjc0Njg2ODEsMjAuNTQ1ODg4OCAxMS42ODQ5Mjc4LDIwLjUyNzIzMDggMTEuNjMwNzk1NywyMC40OTM2NDE2IEw5LjYxMTAxMDc1LDE5LjQzNzkxOTEgTDcuNTkxMjI1ODEsMjAuNDkzNjQxNiBDNy41MzY3MDg2OSwyMC41MjYzNTU3IDcuNDc0OTk4OTUsMjAuNTQ0OTQ0MyA3LjQxMTYxMjksMjAuNTQ3NzQ1NyBDNy4zNTY0NDc1LDIwLjU1MTY0MTcgNy4zMDMwODEyNywyMC41MjcwOTY2IDcuMjY5ODQ5NDYsMjAuNDgyNTQzNCBDNy4yMzgzNzAwMywyMC40MzU4NTU5IDcuMjIyOTAyODMsMjAuMzgwMDcyNSA3LjIyNTgwNjQ1LDIwLjMyMzY5OTQgQzcuMjI2ODY1MDYsMjAuMjkzNzA5MiA3LjIyOTg1MzQyLDIwLjI2MzgxOTggNy4yMzQ3NTI2OSwyMC4yMzQyMTk3IEw3LjYyMTUwNTM4LDE3Ljk5NzIyNTQgTDUuOTgxNTkxNCwxNi40MTM2NDE2IEM1LjkxOTAzODgxLDE2LjM1ODQ2NjYgNS44NzkwNjA1MywxNi4yODE4MyA1Ljg2OTQxOTM1LDE2LjE5ODYxMjcgQzUuODY5NDE5MzUsMTYuMDg4MDkyNSA1Ljk1MzM3NjM0LDE2LjAxODcyODMgNi4xMjEyOTAzMiwxNS45OTA1MjAyIEw4LjM3OTE4MjgsMTUuNjYzODE1IEw5LjM5MTQ4Mzg3LDEzLjYyNzk3NjkgQzkuNDQ3OTEzOTgsMTMuNTA1ODk2IDkuNTIxNTQ4MzksMTMuNDQ0MTYxOCA5LjYxMTY5ODkyLDEzLjQ0NDE2MTggQzkuNzAxODQ5NDYsMTMuNDQ0MTYxOCA5Ljc3NDc5NTcsMTMuNTA1ODk2IDkuODMxOTEzOTgsMTMuNjI3OTc2OSBMMTAuODQ0MjE1MSwxNS42NjM4MTUgTDEzLjEwMjEwNzUsMTUuOTkwNTIwMiBDMTMuMjcwMDIxNSwxNi4wMTg3MjgzIDEzLjM1Mzk3ODUsMTYuMDg4MDkyNSAxMy4zNTM5Nzg1LDE2LjE5ODYxMjcgTDEzLjM1Mzk3ODUsMTYuMTk4NjEyNyBaIiBpZD0iU2hhcGUiIGZpbGw9IiNGRkZGRkYiPjwvcGF0aD4KICAgICAgICAgICAgICAgIDwvZz4KICAgICAgICAgICAgPC9nPgogICAgICAgIDwvZz4KICAgIDwvZz4KPC9zdmc+';
