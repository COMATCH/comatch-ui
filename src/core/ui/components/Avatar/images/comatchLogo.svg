<svg version="1.1" id="BJ7l3Vo-HtM" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 48.1 67.4" style="enable-background:new 0 0 48.1 67.4;">
    <style type="text/css">
        @-webkit-keyframes H1Px2EoZrKM_Animation{0%{opacity: 0.3;}25%{opacity: 1;}75%{opacity: 1;}100%{opacity: 0.3;}}@keyframes H1Px2EoZrKM_Animation{0%{opacity: 0.3;}25%{opacity: 1;}75%{opacity: 1;}100%{opacity: 0.3;}}@-webkit-keyframes S18g3VjbHKG_Animation{0%{opacity: 0.3;}25%{opacity: 1;}75%{opacity: 1;}100%{opacity: 0.3;}}@keyframes S18g3VjbHKG_Animation{0%{opacity: 0.3;}25%{opacity: 1;}75%{opacity: 1;}100%{opacity: 0.3;}}@-webkit-keyframes SyBlnNs-HYG_Animation{0%{opacity: 0.3;}25%{opacity: 1;}75%{opacity: 1;}100%{opacity: 0.3;}}@keyframes SyBlnNs-HYG_Animation{0%{opacity: 0.3;}25%{opacity: 1;}75%{opacity: 1;}100%{opacity: 0.3;}}@-webkit-keyframes BkVl3VsZHKz_Animation{0%{opacity: 0.3;}25%{opacity: 1;}75%{opacity: 1;}100%{opacity: 0.3;}}@keyframes BkVl3VsZHKz_Animation{0%{opacity: 0.3;}25%{opacity: 1;}75%{opacity: 1;}100%{opacity: 0.3;}}#BJ7l3Vo-HtM *{-webkit-animation-duration: 2s;animation-duration: 2s;-webkit-animation-iteration-count: infinite;animation-iteration-count: infinite;-webkit-animation-timing-function: cubic-bezier(0, 0, 1, 1);animation-timing-function: cubic-bezier(0, 0, 1, 1);transform-box: fill-box;-webkit-transform-origin: 50% 50%;transform-origin: 50% 50%;}#BkVl3VsZHKz{fill: #0095b3;-webkit-animation-name: BkVl3VsZHKz_Animation;animation-name: BkVl3VsZHKz_Animation;-webkit-transform-origin: undefined;transform-origin: undefined;}#SyBlnNs-HYG{fill: #0095b3;-webkit-animation-name: SyBlnNs-HYG_Animation;animation-name: SyBlnNs-HYG_Animation;-webkit-transform-origin: undefined;transform-origin: undefined;}#S18g3VjbHKG{fill: #0095b3;-webkit-animation-name: S18g3VjbHKG_Animation;animation-name: S18g3VjbHKG_Animation;-webkit-transform-origin: undefined;transform-origin: undefined;}#H1Px2EoZrKM{fill: #0095b3;-webkit-animation-name: H1Px2EoZrKM_Animation;animation-name: H1Px2EoZrKM_Animation;-webkit-transform-origin: undefined;transform-origin: undefined;}
    </style>
    <path d="M34.7,22.5c5.5,0,9.3-4,9.3-9.6c0-4.6-2.9-9.6-9.3-9.6c-6.4,0-9.3,5-9.3,9.6C25.5,18.6,29.2,22.5,34.7,22.5&#10;&#9;L34.7,22.5z M34.7,7.1c3.3,0,5.1,2.1,5.1,5.8c0,3.7-1.8,5.8-5.1,5.8c-3.3,0-5.1-2.1-5.1-5.8C29.7,9.2,31.5,7.1,34.7,7.1L34.7,7.1z" id="BkVl3VsZHKz"/>
    <polygon points="7.2,24 7.2,60.8 11.6,60.8 11.6,35.4 24.7,48.7 27.5,45.8 7.2,24 " id="SyBlnNs-HYG"/>
    <path d="M14.3,22.5c3,0,5.8-1.4,7.6-3.8l0.1-0.2l-3.1-2.5l-0.1,0.2c-0.9,1.3-2.8,2.3-4.4,2.3c-3.1,0-4.9-2-4.9-5.7&#10;&#9;c0-2.6,0.8-5.7,4.9-5.7c1.6,0,3.5,1,4.4,2.3l0.1,0.2L22,7.2L21.8,7c-1.7-2.4-4.4-3.7-7.6-3.7c-5.5,0-9,3.7-9,9.6&#10;&#9;C5.3,20,10,22.5,14.3,22.5L14.3,22.5z" id="S18g3VjbHKG"/>
    <polygon points="26.3,40.9 29.2,44 37.6,35.4 37.6,60.8 42,60.8 42,24 26.3,40.9 " id="H1Px2EoZrKM"/>
</svg>