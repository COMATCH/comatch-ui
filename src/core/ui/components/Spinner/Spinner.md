
## Spinner

From [`Spinner.js`](Spinner.js)

Displays an animated logo (the COMATCH one by default) to be used as a loading indicator.
The component will fill its container's width.
A size prop is used to set the height (since the container is likely empty, would have no height otherwise)
Size prop can be:
- `page` (default) sets height to 500px
- `panel` sets height to 200px
- a positive number: height value in px

prop | type | default | required | description
---- | :----: | :-------: | :--------: | -----------
**imgSrc** | `String` | `comatchLogoBase64` | :x: | Source of the spinner image
**size** | `Union<Enum('page','panel') \| Number>` | `'page'` | :x: | `page` sizing will take up the whole page whereas `panel` is constrained within a `Panel` component



