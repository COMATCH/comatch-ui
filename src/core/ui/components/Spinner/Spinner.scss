@import '../../../styles/variables.scss';

.Spinner {
  background-color: $co-color-white;
  display: flex;
  justify-content: center;
  align-items: center;

  img {
    max-height: 60%;
    max-width: 82px;
  }
}
