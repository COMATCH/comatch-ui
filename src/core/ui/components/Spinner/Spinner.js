import React from 'react';
import PropTypes from 'prop-types';
import { comatchLogoBase64 } from './images/comatchLogoBase64';

import './Spinner.scss';

/**
 * Displays the animated COMATCH logo to be used as a loading indicator.
 * The component will fill its container's width.
 * A size prop is used to set the height (since the container is likely empty, would have no height otherwise)
 * Size prop can be:
 * - 'page' (default) sets height to 500px
 * - 'panel' sets height to 200px
 * - a positive number: height value in px
 */

const propTypes = {
    imgSrc: PropTypes.string,
    size: PropTypes.oneOfType([PropTypes.oneOf(['page', 'panel']), PropTypes.number]),
};

const defaultProps = {
    imgSrc: comatchLogoBase64,
    size: 'page',
};

function calculateHeight(size) {
    switch (size) {
        case 'page':
            return 500;
        case 'panel':
            return 200;
        default:
            return size;
    }
}

export const Spinner = ({ imgSrc, size }) => (
    <div className="Spinner" style={{ height: calculateHeight(size) }}>
        <img src={imgSrc} alt="spinner icon" />
    </div>
);

Spinner.propTypes = propTypes;

Spinner.defaultProps = defaultProps;
