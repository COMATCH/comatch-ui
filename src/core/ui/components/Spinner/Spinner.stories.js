import React from 'react';
import { storiesOf } from '@storybook/react';
import { Spinner } from './Spinner';

storiesOf('Spinner', module)
    .add('filling the page', () => (
        <div style={{ margin: 10, height: '100%', border: '1px solid navy' }}>
            <Spinner />
        </div>
    ))
    .add('filling a panel', () => {
        const mockPageStyle = {
            page: {
                margin: 10,
                height: '100%',
                display: 'flex',
                flexDirection: 'row',
            },
            main: {
                flex: 4,
                border: '1px solid navy',
                borderRightWidth: 0,
            },
            sidebar: {
                flex: 1,
                display: 'flex',
                flexDirection: 'column',
            },
            sidebarSizedPanel: {
                height: '200px',
                border: '1px solid navy',
                borderBottomWidth: 0,
            },
            sidebarLoadingPanel: {
                minHeight: '20px',
                border: '1px solid navy',
            },
        };

        return (
            <div style={mockPageStyle.page}>
                <div style={mockPageStyle.main} />
                <div style={mockPageStyle.sidebar}>
                    <div style={mockPageStyle.sidebarSizedPanel} />
                    <div style={mockPageStyle.sidebarLoadingPanel}>
                        <Spinner size="panel" />
                    </div>
                </div>
            </div>
        );
    })
    .add('with a small size', () => (
        <div style={{ margin: 10, height: '100%', border: '1px solid navy' }}>
            <Spinner size={80} />
        </div>
    ));
