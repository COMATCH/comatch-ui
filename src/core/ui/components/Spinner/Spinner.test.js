import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { Spinner } from './Spinner';
import '../../../config/tests/setup';
import { comatchLogoBase64 } from './images/comatchLogoBase64';

describe('Spinner tests', () => {
    const DEFAULT_PROPS = {
        imgSrc: comatchLogoBase64,
        size: 'page',
    };

    const OPTIONAL_PROPS = {
        className: 'Spinner',
        onChange: jest.fn(),
        alt: 'spinner icon',
    };

    it('should render and matched values in snapshot', () => {
        const spinnerComponent = renderer.create(<Spinner/>).toJSON();
        expect(spinnerComponent).toMatchSnapshot();
    });

    it('should render correctly with optional props', () => {
        const props = {
            ...OPTIONAL_PROPS,
            ...DEFAULT_PROPS,
        };
        const spinnerComponent = mount(<Spinner {...props} src={comatchLogoBase64}/>);
        expect(spinnerComponent.prop('className')).toEqual(props.className);
        expect(spinnerComponent.prop('onChange')).toEqual(props.onChange);
        expect(spinnerComponent.prop('src')).toEqual(comatchLogoBase64);
        expect(spinnerComponent.prop('alt')).toEqual(props.alt);
        expect(spinnerComponent.prop('imgSrc')).toEqual(comatchLogoBase64);
        expect(spinnerComponent.prop('size')).toEqual(props.size);
    });

    describe('verifying the HTML structure', () => {
        const props = {
            ...OPTIONAL_PROPS,
            ...DEFAULT_PROPS,
        };
        it('should render certain nodes/dom elements', () => {
            const spinnerComponent = mount(<Spinner {...props} />);
            expect(spinnerComponent).toHaveLength(1);

            expect(spinnerComponent.find('div')).toHaveLength(1);
            expect(spinnerComponent.find('div').hasClass('Spinner')).toBe(true);
            expect(spinnerComponent.find('div').props().style).toEqual({ height: 500 });

            expect(spinnerComponent.find('div').find('img')).toHaveLength(1);
            expect(spinnerComponent.find('div').find('img').props().alt).toEqual(props.alt);
        });

        it('should render with `filling a panel`', () => {
            const spinnerComponent = mount(<Spinner {...props} size={'panel'}/>);
            expect(spinnerComponent).toHaveLength(1);
            expect(spinnerComponent.find('div').props().style).toEqual({ 'height': 200 });
        });

        it('should render with `a small size`', () => {
            const spinnerComponent = mount(<Spinner {...props} size={80}/>);
            expect(spinnerComponent).toHaveLength(1);
            expect(spinnerComponent.find('div').props().style).toEqual({ 'height': 80 });
        });
    });
});
