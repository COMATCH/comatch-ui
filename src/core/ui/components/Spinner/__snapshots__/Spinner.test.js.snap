// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner tests should render and matched values in snapshot 1`] = `
<div
  className="Spinner"
  style={
    Object {
      "height": 500,
    }
  }
>
  <img
    alt="spinner icon"
    src="data:image/svg+xml;base64,PHN2ZyB2ZXJzaW9uPSIxLjEiIGlkPSJCSjdsM1ZvLUh0TSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgeD0iMHB4IiB5PSIwcHgiIHZpZXdCb3g9IjAgMCA0OC4xIDY3LjQiIHN0eWxlPSJlbmFibGUtYmFja2dyb3VuZDpuZXcgMCAwIDQ4LjEgNjcuNDsiPiAgICA8c3R5bGUgdHlwZT0idGV4dC9jc3MiPiAgICAgICAgQC13ZWJraXQta2V5ZnJhbWVzIEgxUHgyRW9acktNX0FuaW1hdGlvbnswJXtvcGFjaXR5OiAwLjM7fTI1JXtvcGFjaXR5OiAxO303NSV7b3BhY2l0eTogMTt9MTAwJXtvcGFjaXR5OiAwLjM7fX1Aa2V5ZnJhbWVzIEgxUHgyRW9acktNX0FuaW1hdGlvbnswJXtvcGFjaXR5OiAwLjM7fTI1JXtvcGFjaXR5OiAxO303NSV7b3BhY2l0eTogMTt9MTAwJXtvcGFjaXR5OiAwLjM7fX1ALXdlYmtpdC1rZXlmcmFtZXMgUzE4ZzNWamJIS0dfQW5pbWF0aW9uezAle29wYWNpdHk6IDAuMzt9MjUle29wYWNpdHk6IDE7fTc1JXtvcGFjaXR5OiAxO30xMDAle29wYWNpdHk6IDAuMzt9fUBrZXlmcmFtZXMgUzE4ZzNWamJIS0dfQW5pbWF0aW9uezAle29wYWNpdHk6IDAuMzt9MjUle29wYWNpdHk6IDE7fTc1JXtvcGFjaXR5OiAxO30xMDAle29wYWNpdHk6IDAuMzt9fUAtd2Via2l0LWtleWZyYW1lcyBTeUJsbk5zLUhZR19BbmltYXRpb257MCV7b3BhY2l0eTogMC4zO30yNSV7b3BhY2l0eTogMTt9NzUle29wYWNpdHk6IDE7fTEwMCV7b3BhY2l0eTogMC4zO319QGtleWZyYW1lcyBTeUJsbk5zLUhZR19BbmltYXRpb257MCV7b3BhY2l0eTogMC4zO30yNSV7b3BhY2l0eTogMTt9NzUle29wYWNpdHk6IDE7fTEwMCV7b3BhY2l0eTogMC4zO319QC13ZWJraXQta2V5ZnJhbWVzIEJrVmwzVnNaSEt6X0FuaW1hdGlvbnswJXtvcGFjaXR5OiAwLjM7fTI1JXtvcGFjaXR5OiAxO303NSV7b3BhY2l0eTogMTt9MTAwJXtvcGFjaXR5OiAwLjM7fX1Aa2V5ZnJhbWVzIEJrVmwzVnNaSEt6X0FuaW1hdGlvbnswJXtvcGFjaXR5OiAwLjM7fTI1JXtvcGFjaXR5OiAxO303NSV7b3BhY2l0eTogMTt9MTAwJXtvcGFjaXR5OiAwLjM7fX0jQko3bDNWby1IdE0gKnstd2Via2l0LWFuaW1hdGlvbi1kdXJhdGlvbjogMnM7YW5pbWF0aW9uLWR1cmF0aW9uOiAyczstd2Via2l0LWFuaW1hdGlvbi1pdGVyYXRpb24tY291bnQ6IGluZmluaXRlO2FuaW1hdGlvbi1pdGVyYXRpb24tY291bnQ6IGluZmluaXRlOy13ZWJraXQtYW5pbWF0aW9uLXRpbWluZy1mdW5jdGlvbjogY3ViaWMtYmV6aWVyKDAsIDAsIDEsIDEpO2FuaW1hdGlvbi10aW1pbmctZnVuY3Rpb246IGN1YmljLWJlemllcigwLCAwLCAxLCAxKTt0cmFuc2Zvcm0tYm94OiBmaWxsLWJveDstd2Via2l0LXRyYW5zZm9ybS1vcmlnaW46IDUwJSA1MCU7dHJhbnNmb3JtLW9yaWdpbjogNTAlIDUwJTt9I0JrVmwzVnNaSEt6e2ZpbGw6ICMwMDk1YjM7LXdlYmtpdC1hbmltYXRpb24tbmFtZTogQmtWbDNWc1pIS3pfQW5pbWF0aW9uO2FuaW1hdGlvbi1uYW1lOiBCa1ZsM1ZzWkhLel9BbmltYXRpb247LXdlYmtpdC10cmFuc2Zvcm0tb3JpZ2luOiB1bmRlZmluZWQ7dHJhbnNmb3JtLW9yaWdpbjogdW5kZWZpbmVkO30jU3lCbG5Ocy1IWUd7ZmlsbDogIzAwOTViMzstd2Via2l0LWFuaW1hdGlvbi1uYW1lOiBTeUJsbk5zLUhZR19BbmltYXRpb247YW5pbWF0aW9uLW5hbWU6IFN5QmxuTnMtSFlHX0FuaW1hdGlvbjstd2Via2l0LXRyYW5zZm9ybS1vcmlnaW46IHVuZGVmaW5lZDt0cmFuc2Zvcm0tb3JpZ2luOiB1bmRlZmluZWQ7fSNTMThnM1ZqYkhLR3tmaWxsOiAjMDA5NWIzOy13ZWJraXQtYW5pbWF0aW9uLW5hbWU6IFMxOGczVmpiSEtHX0FuaW1hdGlvbjthbmltYXRpb24tbmFtZTogUzE4ZzNWamJIS0dfQW5pbWF0aW9uOy13ZWJraXQtdHJhbnNmb3JtLW9yaWdpbjogdW5kZWZpbmVkO3RyYW5zZm9ybS1vcmlnaW46IHVuZGVmaW5lZDt9I0gxUHgyRW9acktNe2ZpbGw6ICMwMDk1YjM7LXdlYmtpdC1hbmltYXRpb24tbmFtZTogSDFQeDJFb1pyS01fQW5pbWF0aW9uO2FuaW1hdGlvbi1uYW1lOiBIMVB4MkVvWnJLTV9BbmltYXRpb247LXdlYmtpdC10cmFuc2Zvcm0tb3JpZ2luOiB1bmRlZmluZWQ7dHJhbnNmb3JtLW9yaWdpbjogdW5kZWZpbmVkO30gICAgPC9zdHlsZT4gICAgPHBhdGggZD0iTTM0LjcsMjIuNWM1LjUsMCw5LjMtNCw5LjMtOS42YzAtNC42LTIuOS05LjYtOS4zLTkuNmMtNi40LDAtOS4zLDUtOS4zLDkuNkMyNS41LDE4LjYsMjkuMiwyMi41LDM0LjcsMjIuNSYjMTA7JiM5O0wzNC43LDIyLjV6IE0zNC43LDcuMWMzLjMsMCw1LjEsMi4xLDUuMSw1LjhjMCwzLjctMS44LDUuOC01LjEsNS44Yy0zLjMsMC01LjEtMi4xLTUuMS01LjhDMjkuNyw5LjIsMzEuNSw3LjEsMzQuNyw3LjFMMzQuNyw3LjF6IiBpZD0iQmtWbDNWc1pIS3oiLz4gICAgPHBvbHlnb24gcG9pbnRzPSI3LjIsMjQgNy4yLDYwLjggMTEuNiw2MC44IDExLjYsMzUuNCAyNC43LDQ4LjcgMjcuNSw0NS44IDcuMiwyNCAiIGlkPSJTeUJsbk5zLUhZRyIvPiAgICA8cGF0aCBkPSJNMTQuMywyMi41YzMsMCw1LjgtMS40LDcuNi0zLjhsMC4xLTAuMmwtMy4xLTIuNWwtMC4xLDAuMmMtMC45LDEuMy0yLjgsMi4zLTQuNCwyLjNjLTMuMSwwLTQuOS0yLTQuOS01LjcmIzEwOyYjOTtjMC0yLjYsMC44LTUuNyw0LjktNS43YzEuNiwwLDMuNSwxLDQuNCwyLjNsMC4xLDAuMkwyMiw3LjJMMjEuOCw3Yy0xLjctMi40LTQuNC0zLjctNy42LTMuN2MtNS41LDAtOSwzLjctOSw5LjYmIzEwOyYjOTtDNS4zLDIwLDEwLDIyLjUsMTQuMywyMi41TDE0LjMsMjIuNXoiIGlkPSJTMThnM1ZqYkhLRyIvPiAgICA8cG9seWdvbiBwb2ludHM9IjI2LjMsNDAuOSAyOS4yLDQ0IDM3LjYsMzUuNCAzNy42LDYwLjggNDIsNjAuOCA0MiwyNCAyNi4zLDQwLjkgIiBpZD0iSDFQeDJFb1pyS00iLz48L3N2Zz4="
  />
</div>
`;
